applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp

```diff
@@ -123,8 +123,10 @@ void DistanceModificationProcess::ExecuteInitialize()
 
     if (mContinuousDistance){
         // Continuous distance modification historical variables check
-        const auto& r_node = *mrModelPart.NodesBegin();
-        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(DISTANCE, r_node);
+        if (mrModelPart.GetCommunicator().LocalMesh().NumberOfNodes() != 0) {
+            const auto& r_node = *mrModelPart.NodesBegin();
+            KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(DISTANCE, r_node);
+        }
```

Comment:
May be we can check it in the model part if model part is accessible? 
```suggestion
        KRATOS_ERROR_IF_NOT(mrModelPart.HasNodalSolutionStepVariable(DISTANCE)) << "DISTANCE variable not found in solution step variables list in " << mrModelPart.FullName() << ".\n";
```

Reply:
yep thats better :+1: